forte/data/ontology/core.py

```diff
@@ -494,6 +501,11 @@ def data(self, array: Union[np.ndarray, List]):
 
 
 class BaseLink(Entry, ABC):
+
+    # this type Any is needed since subclasses of this class will have new types
```

Comment:
What's the purpose of adding these two fields here? Does it make them become non-dataclass attribute?